tests/linearGaussian_npse_test.py

```diff
@@ -1,300 +1,226 @@
-        prior_npse = prior if prior_type is None else None
-    elif prior_type == "uniform":
-        prior = BoxUniform(-2 * ones(num_dim), 2 * ones(num_dim))
-        prior_npse = prior
-
-    # This check that our method to handle "general" priors works.
-    # i.e. if NPSE does not get a proper passed by the user.
-    inference = NPSE(prior_npse, show_progress_bars=True, sde_type=sde_type)
-
-    theta = prior.sample((num_simulations,))
-    x = linear_gaussian(theta, likelihood_shift, likelihood_cov)
-
-    score_estimator = inference.append_simulations(theta, x).train(
-        stop_after_epochs=200
-    )
-
-    return {
-        "score_estimator": score_estimator,
-        "inference": inference,
-        "prior": prior,
-        "likelihood_shift": likelihood_shift,
-        "likelihood_cov": likelihood_cov,
-        "prior_mean": prior_mean
-        if prior_type == "gaussian" or prior_type is None
-        else None,
-        "prior_cov": prior_cov
-        if prior_type == "gaussian" or prior_type is None
-        else None,
-        "num_dim": num_dim,
-    }
+    return inference, score_estimator, test_case
 
 
 @pytest.mark.slow
-@pytest.mark.parametrize(
-    "iid_method, num_trial",
-    [
-        pytest.param("fnpe", 3, id="fnpe-2trials"),
-        pytest.param("gauss", 3, id="gauss-6trials"),
-        pytest.param("auto_gauss", 8, id="auto_gauss-8trials"),
-        pytest.param("auto_gauss", 16, id="auto_gauss-16trials"),
-        pytest.param("jac_gauss", 8, id="jac_gauss-8trials"),
-    ],
-)
-def test_npse_iid_inference(
-    npse_trained_model, iid_method, num_trial, sde_type, prior_type
-):
-    """Test whether NPSE infers well a simple example with available ground truth."""
-    num_samples = 1000
-
-    # Extract data from fixture
-    score_estimator = npse_trained_model["score_estimator"]
-    inference = npse_trained_model["inference"]
-    prior = npse_trained_model["prior"]
-    likelihood_shift = npse_trained_model["likelihood_shift"]
-    likelihood_cov = npse_trained_model["likelihood_cov"]
-    prior_mean = npse_trained_model["prior_mean"]
-    prior_cov = npse_trained_model["prior_cov"]
-    num_dim = npse_trained_model["num_dim"]
-
-    x_o = zeros(num_trial, num_dim)
-    posterior = inference.build_posterior(score_estimator)
-    posterior.set_default_x(x_o)
-    samples = posterior.sample((num_samples,), iid_method=iid_method)
+@pytest.mark.parametrize("sample_with", sample_with)
+@pytest.mark.parametrize("iid_method", iid_methods)
+@pytest.mark.parametrize("npse_trained_model", test_cases_all, indirect=True)
+def test_c2st(npse_trained_model, iid_method, sample_with):
+    num_samples = 1_000
+    inference, score_estimator, test_case = npse_trained_model
+    num_dim = test_case.num_dim
+    sde_type = test_case.sde_type
+    prior_str = test_case.prior_type
+
+    posterior = inference.build_posterior(score_estimator, sample_with=sample_with)
+    posterior.set_default_x(test_case.x_o)
```

Comment:
how are the number of trials specified here? 
the GT posterior will change for every different number of trials

Reply:
I am not sure how it depends on the number of trials. Could you please clarify this?